### PR TITLE
Add notification category endpoints and types

### PR DESCRIPTION
**Pull request by shahnami**
This PR adds notification category endpoints and typings.

### CODE REVIEW

File: packages/sentinel/src/api/index.ts [hunk 1]
```diff
@@ -250,12 +289,16 @@ export class SentinelClient extends BaseApiClient {
...
+    const notificationChannels = await this.getNotifications(sentinel.notificationChannels);
+
     return {
       ...partialResponse,
       name: sentinel.name,
       alertThreshold: sentinel.alertThreshold,
       notifyConfig: {
-        notifications: await this.getNotifications(sentinel.notificationChannels),
+        notifications: notificationChannels,
+        // only allow to set category if channels is empty
```
Comment (pasevin) [inline packages/sentinel/src/api/index.ts:300]: Nit: no need for this comment imho
```suggestion
```

File: packages/sentinel/src/api/index.ts [hunk 2]
```diff
@@ -290,6 +333,9 @@ export class SentinelClient extends BaseApiClient {
       alertTimeoutMs: sentinel.notifyConfig?.timeoutMs,
       alertMessageBody: sentinel.notifyConfig?.messageBody,
       notificationChannels: sentinel.notifyConfig?.notifications?.map(({ notificationId }) => notificationId) ?? [],
+      // notificationCategoryId will be replaced with undefined in `constructSentinelRequest`
+      // if notification channels is not empty
```
Comment (pasevin) [inline packages/sentinel/src/api/index.ts:337]: Nit: Do we need this comment?
```suggestion
```

File: packages/sentinel/src/api/index.ts [hunk 3]
```diff
@@ -306,6 +352,9 @@ export class SentinelClient extends BaseApiClient {
...
```
Comment (pasevin) [inline packages/sentinel/src/api/index.ts:356]: Nit: Do we need this comment?
```suggestion
```

**Review by pasevin — APPROVED** [review]
LGTM!

**Review by mok0230 — APPROVED** [review]
LGTM!

**Pull request merged**